pkg/web/apps/apps_http_handler.go

```diff
@@ -34,20 +35,43 @@ func NewHTTPHandler(e *echo.Echo, s Service) HTTPHandler {
 
 // GetApps returns all apps as JSON from the AppService
 func (a *httpHandler) GetApps(c echo.Context) error {
-	apps, err := a.Service.GetApps()
+	hasAppId := len(c.QueryParam("appId")) > 0
+	apps, err := a.HandleGetApp(c)
 
 	// If no apps have been found, return a HTTP Status code of 204 with no response body
-	if err == models.ErrNotFound {
+	if err == models.ErrNotFound && !hasAppId {
 		return c.NoContent(http.StatusNoContent)
 	}
 
+	// If the appId was informed than it should return 404
+	if err == models.ErrNotFound && hasAppId {
+		return c.NoContent(http.StatusNotFound)
+	}
```

Comment:
it is not a big deal, but I think this change is unnecessary to return a different status code when there is an app in in the request. It gives more information away about the system for no obvious benefit. In the worst scenario, it means someone could use this information to guess the appid in the system. It's probably doesn't matter that much for this system, and it actually makes the code more complicated. Given that this wasn't there, I would suggest keep it the old way.

Reply:
@wei-lee all project is returning 404 when the app is not found. For the app is not found it does not exist and/or is with the deleted_at <> NULL. I am just following how it was defined to get done.

Reply:
I would agree with Wei on this, it makes sense to leave it as just

```
if err == models.ErrNotFound {
	return c.NoContent(http.StatusNoContent)
}
```

Reply:
@davidffrench and all other places that are returning 404 when the app is not found and or have the deleted_at <> NULL?

Reply:
I don't mind if it is a 204 or 404. I think 404 is confusing in the context of an empty REST collection.

The main point of Weis comment was that there is a different response status code sent for when a model is not found and the appId filter is sent as a query param and when it is not. This should not be the case, the response code should be consistent for this single endpoint when no resources were found.